client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java

```diff
@@ -693,6 +693,7 @@ private Boolean handleScaleResponse(Stream stream, ScaleResponse response) {
             log.debug("Received the following data from the controller {}", ranges.getSegmentRangesList());
             NavigableMap<Double, Segment> rangeMap = new TreeMap<>();
             for (SegmentRange r : ranges.getSegmentRangesList()) {
+                Preconditions.checkState(r.getMinKey() <= r.getMaxKey());
```

Comment:
We could add an appropriate error message for the added ` Preconditions.checkState` and `Preconditions.checkArgument`